ui/static_manifest.go

```diff
@@ -44,12 +44,17 @@ func (h *handler) showWebManifest(w http.ResponseWriter, r *http.Request) {
 		BackgroundColor string                 `json:"background_color"`
 	}
 
+	user, err := h.store.UserByID(request.UserID(r))
```

Comment:
The `/manifest.json` endpoint can be accessed when the user is not authenticated. That part will panic.

```
curl http://127.0.0.1:8080/manifest.json
```

```
[2021-02-01T20:07:10] [DEBUG] [HTTP] 127.0.0.1 GET /manifest.json
[2021-02-01T20:07:10] [DEBUG] [UI:AppSession] Session not found, creating a new one
2021/02/01 20:07:10 http: panic serving 127.0.0.1:53475: runtime error: invalid memory address or nil pointer dereference
goroutine 67 [running]:
net/http.(*conn).serve.func1(0xc000856000)
	/usr/local/go/src/net/http/server.go:1801 +0x147
panic(0x1730260, 0x1de5450)
	/usr/local/go/src/runtime/panic.go:975 +0x47a
miniflux.app/ui.(*handler).showWebManifest(0xc0008b3c20, 0x19516e0, 0xc00058a3c0, 0xc000843a00)
	/Users/fred/repos/miniflux/app/v2/ui/static_manifest.go:57 +0x44d
...
```

Reply:
I did not notice that, sorry. I added a few lines to retrieve the user's preferred display mode *only if* the user is authenticated.

I am not sure if it's okay to fix it that way, at least from a design/architectural standpoint. For example, I see there are other user preferences that are also stored in context (e.g. timezone, language, theme).